engine/src/main/java/org/hibernate/search/v6poc/search/dsl/predicate/impl/NestedPredicateContextImpl.java

```diff
@@ -46,7 +46,7 @@
 
 	@Override
 	public void contribute(CTX context, C collector) {
-		singlePredicateContributor.contribute( context, builder.getNestedCollector() );
+		builder.getNestedCollector().accept( singlePredicateContributor );
```

Comment:
This should be done directly when you receive the `singlePredicateContributor`. Otherwise you run the risk of adding it twice to the builder, which is precisely what we wanted to avoid in the boolean predicate context in particular.

I know this probably works as it is in this particular case, but it's only because the consumer we're calling does not check it's only been called twice, and erases the previous value whenever it's called a second time. Two wrongs make a right, if you will :)

Reply:
RIght, done.